faunadb/query.py

```diff
@@ -40,6 +43,43 @@ def quote(expr):
   """See the `docs <https://faunadb.com/documentation/queries#basic_forms>`__."""
   return {"quote": expr}
 
+thread_local = local()
+thread_local.lambda_auto_var_number = 0
```

Comment:
Also we called this `fauna_lambda_var_number` in ruby, I think that's a more properly scoped name than this one.